### PR TITLE
[Console] HHVM read input stream bug

### PR DESCRIPTION
**Pull request by mbutkereit**
| Q | A |
| --- | --- |
| Bug fix? | [yes] |
| New feature? | [no] |
| BC breaks? | [no] |
| Deprecations? | [no] |
| Tests pass? | [yes] |
| License | MIT |

HHVM readline() function requires a parameter  see:

https://github.com/facebook/hhvm/blob/master/hphp/runtime/ext/readline/ext_readline.php

But the php readline does not require one.

```
In   Symfony\Component\Console\Helper\QuestionHelper
 …
 readFromInput($stream)
 …
```

we use readline() without a parameter.

HHVM Version:

```
HipHop VM 3.10.1 (rel)
Compiler: tags/HHVM-3.10.1-0-g689b4969a141620ee5a282ce0dbf72278c84d44b
Repo schema: 6c99ee1f98340f6f3ef397a332583f0e843a627d
```

Docker Container: 

```
docker run --rm -it -v `pwd`:`pwd` -w `pwd` brunoric/hhvm:deb hhvm test.php ls 
```

Test.php

```
<?php
use Symfony\Component\Console\Application;
use Symfony\Component\Console\Input\InputInterface;
use Symfony\Component\Console\Input\InputArgument;
use Symfony\Component\Console\Input\InputOption;
use Symfony\Component\Console\Output\OutputInterface;
use Symfony\Component\Console\Question\Question;

use Symfony\Component\Console\Helper\QuestionHelper;

require_once  "vendor/autoload.php";

$console = new Application();

$console
    ->register('ls')
    ->setDescription('Try input')
    ->setCode(function (InputInterface $input, OutputInterface $output) {

        $question = new Question('Your email address: ');
        $question->setValidator(
            function ($answer) {
                return $answer;
            }
        );
        $question->setMaxAttempts(5);
            $helper = new QuestionHelper();
          $email = $helper->ask($input, $output, $question);

        $output->writeln(sprintf('Email Adress: <info>%s</info>', $email));
    })
;
$console->run();
```

composer.json

```
{
    "require": {
        "symfony/console": "^3.0"
    }
}
```

Result:
No input is possible and you got a warning 

```
Your email address: 
Warning: readline() expects exactly 1 parameter, 0 given in /clitest/vendor/symfony/console/Helper/QuestionHelper.php on line 436
Email Adress:
```

Fix:
Add a empty string parameter to  readline() or use fgets() for hhvm.

### CODE REVIEW

**Comment by xabbuh** [comment]
Can you add a test to avoid regressions?

File: src/Symfony/Component/Console/Helper/QuestionHelper.php [hunk 1]
```diff
@@ -437,7 +437,7 @@ private function getShell()
      */
     private function readFromInput($stream)
     {
-        if (STDIN === $stream && function_exists('readline')) {
+        if (STDIN === $stream && function_exists('readline') && !defined('HHVM_VERSION')) {
             $ret = readline();
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Helper/QuestionHelper.php]: giving the empty string looks more feature-full, isn't it? why prefer fgets on hhvm when readline works?

**Comment by mbutkereit** [comment]
@nicolas-grekas  i think you are right. (i have changed it)
@xabbuh  i dont' know how i can write a test for readline().
 Does it make sense to test stdin( readline() ) ?  i don't know.

**Comment by stof** [comment]
Please also report it to HHVM as an incompatibility bug.

**Comment by mbutkereit** [comment]
@stof done see https://github.com/facebook/hhvm/issues/6731

**Comment by nicolas-grekas** [comment]
:+1:
Status: reviewed

**Comment by nicolas-grekas** [comment]
Thank you @mbutkereit.

**Pull request closed without merging**